rust/src/util/api.rs

```diff
@@ -32,16 +33,30 @@ pub fn init_log_with_file(file: File) -> Option<()> {
     }
 }
 
+// Serialize the GPU device names into a vector
+#[cfg(any(feature = "opencl", feature = "cuda"))]
+fn get_gpu_devices_internal() -> Vec<slice_boxed<u8>> {
+    let devices = rust_gpu_tools::Device::all();
+    let devices: Vec<_> = devices
+        .into_iter()
+        .map(|d| d.name().into_bytes().into_boxed_slice().into())
+        .collect();
+
+    devices
```

Comment:
The less code version without rebinding the variable would be:

```suggestion
    devices
        .into_iter()
        .map(|d| d.name().into_bytes().into_boxed_slice().into())
        .collect()
```